src/Juvix/Compiler/Asm/Translation/FromCore.hs

```diff
@@ -252,9 +283,62 @@ genCode infoTable fi =
     snocPopTemp False code = DL.snoc code (mkInstr PopTemp)
     snocPopTemp True code = code
 
-    convertType :: Int -> Core.Type -> Type
-    convertType = unimplemented
-
 -- Be mindful that JuvixAsm types are explicitly uncurried, while
--- Core.Stripped types are always curried. If a function takes `n` arguments
+-- Core.Stripped types are always curried. If a function takes `n` arguments,
 -- then the first `n` arguments should be uncurried in its JuvixAsm type.
+convertType :: Int -> Core.Type -> Type
+convertType argsNum ty =
+  case ty of
+    Core.TyDynamic ->
+      TyDynamic
+    Core.TyPrim (Core.PrimInteger Core.PrimIntegerInfo {..}) ->
```

Comment:
i'd prefer to have a nested case for `Core.TyPrim`

Reply:
Done